### PR TITLE
Align the documentation with new API

### PR DESCRIPTION
**Pull request by K-Jo**
#51

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -1,68 +1,103 @@
...
-* aggType - aggregation type one of the following: avg, sum, min, max, range, count, first, last
-* bucketSizeSec - time bucket for aggregated compaction,
-* destKey - key name for destination time series
-
-> DEST_KEY should be of a `timeseries` type, and should be created before TS.CREATERULE is called.
+* timestamp - unix timestamp (in seconds) or `*` for automatic timestamp (using the system clock)
+* value - sample numeric data value (double)
 
-> *Performance Notice*: if a compaction rule exits on a timeseries `TS.ADD` performance might be reduced, the complexity of `TS.ADD` is always O(M) when M is the amount of compactions rules or O(1).
+Optional args:
```
Comment (gkorland) [inline docs/commands.md]: Add a comment why those args are really needed here or point the notes bellow

Comment (gkorland) [reply docs/commands.md]: Are they going to override the original one set in CREATE?

File: docs/commands.md [hunk 2]
```diff
@@ -1,68 +1,103 @@
...
-TS.DELETERULE sourceKey destKey
+TS.ADD temperature:2:32 1548149180 26 LABELS sensor_id 2 area_id 32 
+TS.ADD temperature:3:11 1548149180 27 RETENTION 3600
+TS.ADD temperature:3:11 1548149181 30
 ```
 
-* sourceKey - key name for source time series
-* destKey - key name for destination time series
+#### Complexity
+If a compaction rule exits on a timeseries `TS.ADD` performance might be reduced, the complexity of `TS.ADD` is always O(M) when M is the amount of compactions rules or O(1).
```
Comment (gkorland) [inline docs/commands.md]: or O(1)?

File: docs/commands.md [hunk 3]
```diff
@@ -1,68 +1,103 @@
...
+TS.DECRBY key value [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
+```
+This command can be used as a counter/gauge that get automatic history as a time series.
+
+* key - key name for timeseries
 * value - sample numeric data value (double)
 
-### Examples
+Optional args:
+   * time-bucket - time bucket for resetting the current counter in seconds
```
Comment (gkorland) [inline docs/commands.md]: resetting

File: docs/commands.md [hunk 4]
```diff
@@ -1,68 +1,103 @@
...
-TS.ADD temperature:3:11 1548149181 30
+TS.CREATERULE sourceKey destKey AGGREGATION aggType bucketSizeSeconds
+```
+* sourceKey - key name for source time series
+* destKey - key name for destination time series
+* aggType - aggregation type one of the following: avg, sum, min, max, range, count, first, last
+* bucketSizeSeconds - time bucket for aggregation in seconds
+
+> DEST_KEY should be of a `timeseries` type, and should be created before TS.CREATERULE is called.
+
```
Comment (gkorland) [inline docs/commands.md]: Example?

File: docs/commands.md [hunk 5]
```diff
@@ -100,22 +135,22 @@ This can be improved in the future by using binary search to find the start of t
...
 ```
 
-* labels - set of key-pair selector (`k=v`, `k!=v,` `k=` contains a key, `k!=` doesn't contain a key)
 * fromTimestamp - start timestamp for range query
 * toTimestamp - end timestamp for range query
 
 Optional args:
    * aggregationType - one of the following: avg, sum, min, max, count, first, last
    * bucketSizeSeconds - time bucket for aggregation in seconds
+   * filters - set of key-pair fitlers (`k=v`, `k!=v,` `k=` contains a key, `k!=` doesn't contain a key)
```
Comment (gkorland) [inline docs/commands.md]: if Filter optional?

File: docs/commands.md [hunk 6]
```diff
@@ -100,22 +135,22 @@ This can be improved in the future by using binary search to find the start of t
    2) "20"
 
 ```
-## TS.RANGEBYLABELS - ranged query by labels
+## TS.MRANGE - ranged query by filters
 ```sql
-TS.RANGEBYLABELS (labels) fromTimestamp toTimestamp [aggregationType] [bucketSizeSeconds]
+TS.MRABGE fromTimestamp toTimestamp [AGGREGATION aggregationType bucketSizeSeconds] FILTER filter..
```
Comment (gkorland) [inline docs/commands.md]: "MRABGE"?

File: docs/commands.md [hunk 7]
```diff
@@ -100,22 +135,22 @@ This can be improved in the future by using binary search to find the start of t
...
 Optional args:
    * aggregationType - one of the following: avg, sum, min, max, count, first, last
    * bucketSizeSeconds - time bucket for aggregation in seconds
+   * filters - set of key-pair fitlers (`k=v`, `k!=v,` `k=` contains a key, `k!=` doesn't contain a key)
 
-### Examples
+### Example
 ```sql
-127.0.0.1:6379> TS.RANGEBYLABELS area_id=32 1548149180 1548149210 avg 5
+127.0.0.1:6379> TS.MRANGE 1548149180 1548149210 AGGREGATION avg 5 FILTER area_id=32
```
Comment (gkorland) [inline docs/commands.md]: Extend the example to have two filter of different types e.g. >=

File: docs/commands.md [hunk 8]
```diff
@@ -154,67 +189,17 @@ Optional args:
...
-```sql
-TS.INCRBY key value [RESET time-bucket]
-```
-or
+## General
+
+### TS.INFO - query the series metadata
 ```sql
-TS.DECRBY key value [RESET time-bucket]
+TS.INFO key
```
Comment (gkorland) [inline docs/commands.md]: add a list of metadata available

File: docs/commands.md (hunk 1 above)
Comment (danni-m) [reply docs/commands.md]: @gkorland No, they are used if the key doesn't exists.

File: docs/commands.md [hunk 9]
```diff
@@ -1,68 +1,103 @@
...
-TS.CREATERULE sourceKey aggType bucketSizeSec destKey
+TS.ADD key timestamp value [RETENTION retentionSecs] [LABELS field value..]
 ```
-* sourceKey - key name for source time series
-* aggType - aggregation type one of the following: avg, sum, min, max, range, count, first, last
-* bucketSizeSec - time bucket for aggregated compaction,
-* destKey - key name for destination time series
-
-> DEST_KEY should be of a `timeseries` type, and should be created before TS.CREATERULE is called.
+* timestamp - unix timestamp (in seconds) or `*` for automatic timestamp (using the system clock)
```
Comment (danni-m) [inline docs/commands.md]: timestamp is a 64bit unsigned integer (so not just secs)

File: docs/commands.md (hunk 2 above)
Comment (danni-m) [reply docs/commands.md]: Lets make it simple: O(1) without rules and O(M) with rules

File: docs/commands.md [hunk 10]
```diff
@@ -1,68 +1,103 @@
...
+TS.CREATERULE sourceKey destKey AGGREGATION aggType bucketSizeSeconds
+```
+* sourceKey - key name for source time series
+* destKey - key name for destination time series
+* aggType - aggregation type one of the following: avg, sum, min, max, range, count, first, last
+* bucketSizeSeconds - time bucket for aggregation in seconds
+
+> DEST_KEY should be of a `timeseries` type, and should be created before TS.CREATERULE is called.
+
+> *Performance Notice*: if a compaction rule exits on a timeseries `TS.ADD` performance might be reduced, the complexity of `TS.ADD` is always O(M) when M is the amount of compactions rules or O(1).
```
Comment (danni-m) [inline docs/commands.md]: This notice should be kept in TS.ADD IMO.

File: docs/commands.md (hunk 5 above)
Comment (danni-m) [reply docs/commands.md]: Filter is not optional.

File: docs/commands.md (hunk 10 above)
Comment (K-Jo) [reply docs/commands.md]: removed it here

File: docs/commands.md (hunk 8 above)
Comment (K-Jo) [reply docs/commands.md]: There's an example output with this, isn't this sufficient?

File: docs/commands.md (hunk 7 above)
Comment (K-Jo) [reply docs/commands.md]: I added a !=

File: docs/commands.md (hunk 3 above)
Comment (K-Jo) [reply docs/commands.md]: Don't get it, do you want to clarify that it goes to 0?

File: docs/commands.md (hunk 1 above)
Comment (K-Jo) [reply docs/commands.md]: but they are updated too if they differ from existing values right?

File: docs/commands.md (hunk 9 above)
Comment (K-Jo) [reply docs/commands.md]: @danni-m so milliseconds?

File: docs/commands.md (hunk 1 above)
Comment (danni-m) [reply docs/commands.md]: @K-Jo not at this version.

**Pull request merged**